src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/controller/TriggerProcessingCrnController.kt

```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.digital.hmpps.hmppsallocations.controller
+
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.reactive.asFlow
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.http.codec.multipart.FilePart
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestPart
+import org.springframework.web.bind.annotation.RestController
+import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.hmppsallocations.service.TriggerReprocessService
+
+@RestController
+@RequestMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
+class TriggerProcessingCrnController(private val triggerReprocessService: TriggerReprocessService) {
+
+  @PostMapping("/crn/upload")
```

Comment:
`upload` is the wrong name IMO - `refresh` or `reprocess` would be a better reflection of what it does.